src/components/TradesWidget/index.tsx

```diff
@@ -1,20 +1,119 @@
-import React from 'react'
+import React, { useCallback, useMemo } from 'react'
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faFileCsv } from '@fortawesome/free-solid-svg-icons'
+
+import { formatPrice, TokenDetails, formatAmount } from '@gnosis.pm/dex-js'
 
 import { ContentPage } from 'components/Layout/PageWrapper'
 import { CardTable } from 'components/Layout/Card'
+import { FileDownloaderLink } from 'components/FileDownloaderLink'
 
 import { useWalletConnection } from 'hooks/useWalletConnection'
 import { useTrades } from 'hooks/useTrades'
 
-import { TradeRow } from 'components/TradesWidget/TradeRow'
+import { Trade } from 'api/exchange/ExchangeApi'
+
+import { toCsv } from 'utils/csv'
+
+import { TradeRow, classifyTrade } from 'components/TradesWidget/TradeRow'
+import { getNetworkFromId, isTradeSettled } from 'utils'
+
+const csvHeaders = [
+  'Market',
+  'BuyTokenSymbol',
+  'BuyTokenAddress',
+  'SellTokenSymbol',
+  'SellTokenAddress',
+  'Limit Price',
+  'Fill Price',
+  'Amount',
+  'Received',
+  'Type',
+  'Time',
+  'TransactionHash',
+  'EventLogIndex',
+  'OrderId',
+  'BatchId',
+]
+
+function symbolOrAddress(token: TokenDetails): string {
+  return token.symbol || token.address
+}
+
+function csvTransformer(trade: Trade): string[] {
+  const {
+    buyToken,
+    sellToken,
+    limitPrice,
+    fillPrice,
+    sellAmount,
+    buyAmount,
+    timestamp,
+    txHash,
+    eventIndex,
+    orderId,
+    batchId,
+  } = trade
+
+  return [
+    `${symbolOrAddress(buyToken)}/${symbolOrAddress(sellToken)}`,
+    buyToken.symbol || '',
+    buyToken.address,
+    sellToken.symbol || '',
+    sellToken.address,
+    formatPrice(limitPrice),
+    formatPrice(fillPrice),
+    formatAmount({
+      amount: sellAmount,
+      precision: sellToken.decimals as number,
+      decimals: sellToken.decimals,
+      thousandSeparator: false,
+      isLocaleAware: false,
+    }),
+    formatAmount({
+      amount: buyAmount,
+      precision: buyToken.decimals as number,
+      decimals: sellToken.decimals,
+      thousandSeparator: false,
+      isLocaleAware: false,
+    }),
+    classifyTrade(trade),
+    new Date(timestamp).toISOString(),
+    txHash,
+    eventIndex.toString(),
+    orderId,
+    batchId.toString(),
+  ]
+}
 
 const Trades: React.FC = () => {
-  const { networkId } = useWalletConnection()
+  const { networkId, userAddress } = useWalletConnection()
   const trades = useTrades()
 
+  const generateCsv = useCallback(
+    () =>
+      toCsv({
+        headers: csvHeaders,
+        data: trades.filter(isTradeSettled),
+        transformer: csvTransformer,
+      }),
+    [trades],
+  )
+
+  const filename = useMemo(
+    () => `trades_${getNetworkFromId(networkId as number).toLowerCase()}_${userAddress}_${new Date().getTime()}.csv`,
```

Comment:
Does

```suggestion
    () => `trades_${getNetworkFromId(networkId!).toLowerCase()}_${userAddress}_${new Date().getTime()}.csv`,
```

work? It's less of a hack than `as` type cast

Reply:
![screenshot_2020-05-29_14-17-49](https://user-images.githubusercontent.com/43217/83306270-1f091f00-a1b7-11ea-8995-82ecc773ab55.png)

Either a cast or TS eslint warning